### PR TITLE
refactor: remove several files in priorities package

### LINKED ISSUE #86994: Remove reduce.go, metadata.go and types.go from priority package
/sig scheduling
/priority important-soon
/help
part of #85822

Note that #86890 should be merged first.

### PR DESCRIPTION
**Pull request by draveness**
/sig scheduling
/priority important-soon
/kind cleanup

Fixes #86994

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/serviceaffinity/service_affinity_test.go [hunk 1]
```diff
@@ -380,11 +377,11 @@ func TestServiceAffinityScore(t *testing.T) {
 			name: "three pods, two service pods, with rack label",
 		},
 	}
-	// these local variables just make sure controllerLister\replicaSetLister\statefulSetLister not nil
-	// when construct metaDataProducer
-	sss := []*apps.StatefulSet{{Spec: apps.StatefulSetSpec{Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}}}}
-	rcs := []*v1.ReplicationController{{Spec: v1.ReplicationControllerSpec{Selector: map[string]string{"foo": "bar"}}}}
-	rss := []*apps.ReplicaSet{{Spec: apps.ReplicaSetSpec{Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"foo": "bar"}}}}}
+	// // these local variables just make sure controllerLister\replicaSetLister\statefulSetLister not nil
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/serviceaffinity/service_affinity_test.go]: Delete commented out code

**Comment by ahg-g** [comment]
Can you also move all constants in [priorities.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/priorities/priorities.go#L19) to ```framework/plugins/legacy_registry.go``` and delete priorities.go please.

**Comment by draveness** [comment]
> Can you also move all constants in [priorities.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/algorithm/priorities/priorities.go#L19) to `framework/plugins/legacy_registry.go` and delete priorities.go please.

sure

File: pkg/scheduler/framework/plugins/legacy_registry.go [hunk 2]
```diff
@@ -47,7 +46,47 @@ import (
...
+	// requested by the pod present.
+	ImageLocalityPriority = "ImageLocalityPriority"
+	// ResourceLimitsPriority defines the nodes of prioritizer function ResourceLimitsPriority.
+	ResourceLimitsPriority = "ResourceLimitsPriority"
+	// EvenPodsSpreadPriority defines the name of prioritizer function that prioritizes nodes
+	// which have pods and labels matching the incoming pod's topologySpreadConstraints.
+	EvenPodsSpreadPriority = "EvenPodsSpreadPriority"
+)
+
+// LegacyRegistry is used to store current state of registered predicates and
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/legacy_registry.go]: nit: commend should end with "and priorities."

File: pkg/scheduler/framework/plugins/legacy_registry.go [hunk 3]
```diff
@@ -82,7 +121,7 @@ type ConfigProducerArgs struct {
 // predicate/priority given the args.
 type ConfigProducer func(args ConfigProducerArgs) (config.Plugins, []config.PluginConfig)
 
-// NewLegacyRegistry returns a legacy algorithm registry of predicates and priorities.
+// NewLegacyRegistry returns a legacy algorithm registry of predicates and
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/legacy_registry.go]: ditto

File: pkg/scheduler/framework/plugins/legacy_registry.go [hunk 4]
```diff
@@ -228,52 +267,52 @@ func NewLegacyRegistry() *LegacyRegistry {
 			return
 		})
 
-	// Register Priorities.
-	registry.registerPriorityConfigProducer(priorities.SelectorSpreadPriority,
+	// Register
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/legacy_registry.go]: ditto

File: pkg/scheduler/framework/plugins/legacy_registry.go [hunk 5]
```diff
@@ -467,7 +506,7 @@ func (lr *LegacyRegistry) ProcessPriorityPolicy(policy config.PriorityPolicy, co
 	}
 
 	if policy.Argument.ServiceAntiAffinity != nil {
-		// We use the ServiceAffinity plugin name for all ServiceAffinity custom priorities.
+		// We use the ServiceAffinity plugin name for all ServiceAffinity custom
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/legacy_registry.go]: ditto

File: pkg/scheduler/framework/plugins/legacy_registry.go [hunk 6]
```diff
@@ -482,7 +521,7 @@ func (lr *LegacyRegistry) ProcessPriorityPolicy(policy config.PriorityPolicy, co
 	}
 
 	if policy.Argument.LabelPreference != nil {
-		// We use the NodeLabel plugin name for all NodeLabel custom priorities.
+		// We use the NodeLabel plugin name for all NodeLabel custom
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/legacy_registry.go]: ditto

**Comment by draveness** [comment]
> nit: commend should end with "and priorities."

Oops, thanks for the catch, I used `s/priorities.//g` to update the file...

**Comment by ahg-g** [comment]
Thanks, https://github.com/kubernetes/kubernetes/pull/86979 just got merged, please rebase.

/approve

**Comment by draveness** [comment]
Rebased the master, please take a look

**Pull request merged**